libraries/chain/controller.cpp

```diff
@@ -712,27 +732,51 @@ struct building_block {
                         // Construct a default QC claim.
                         qc_data = qc_data_t{ {}, bb.parent.core.latest_qc_claim() };
                      }
-                  });
 
+                     finality_mroot_claim = get_finality_mroot_claim(bb.parent, qc_data->qc_claim);
+                  });
                }
```

Comment:
I thought you were going to create a method for the above that returned the `qc_data` and `finality_mroot_claim` to avoid the use of the optionals.